### ISSUE TITLE
feat: cloud training pipeline — agent tool for fine-tuning on EC2/SageMaker

### ISSUE THREAD

**Issue by mikegc-aws** [strands-labs/robots#138]
## Summary

The library handles data collection (`DatasetRecorder`, `lerobot_teleoperate`) and inference (`gr00t_inference`, `lerobot_local`) but the critical middle step — fine-tuning a model on collected demonstrations — is completely absent. For real hardware users, the pretrained base model doesn't produce useful behavior without fine-tuning on demonstrations from their specific setup (cameras, workspace, calibration). This is the gap between a sim demo and a working real robot.

Since this is an AWS-backed project, cloud GPU training is a natural fit. Most users with desktop arms (SO-100, Koch, ViperX) won't have local NVIDIA GPUs capable of fine-tuning (L40S 48GB minimum for GR00T at batch 32).

## Motivation

**Today's user journey breaks at step 3:**

1. `Robot("so100")` — connect hardware ✅
2. `agent("Record 50 demos of picking up the red cube")` — collect data ✅
3. `agent("Train a model on those demos")` — ❌ **nothing happens**
4. `agent("Pick up the red cube")` — needs a fine-tuned model

From hands-on experience building this pipeline manually, training involves:
- Provisioning a GPU instance (g6e.xlarge / L40S 48GB minimum — A10G 24GB OOMs even at batch 1)
- Converting LeRobot v3 dataset → v2.1 format (schema transformation)
- Injecting `modality.json` mapping camera/state/action keys to GR00T's expected structure
- Adding `annotation.human.task_description` column to parquet files
- Running Isaac-GR00T `launch_finetune.py` with correct flags (`--embodiment-tag NEW_EMBODIMENT`)
- Uploading the ~12GB checkpoint to S3
- Deploying to inference server

Each of these steps has non-obvious gotchas:
- `flash-attn` requires `pip install wheel` first and `--no-build-isolation`
- The `isaac-gr00t-omniverse` pip URL 404s — must clone from `github.com/NVIDIA/Isaac-GR00T`
- NVMe ephemeral storage on EC2 is lost on instance stop — must sync checkpoints to S3
- `--embodiment-tag NEW_EMBODIMENT` is critical but undocumented for novel robots

## Proposal

A `training` agent tool (or set of tools) that wraps this workflow:

```python
from strands_robots import Robot
from strands_robots.training import TrainingTool
from strands import Agent

robot = Robot("so100")
training = TrainingTool(
    compute="aws",              # or "local" for Jetson/workstation
    instance_type="g6e.xlarge", # auto-selected if omitted
    checkpoint_bucket="s3://my-bucket/checkpoints/",
)
agent = Agent(tools=[robot, training])

# The agent can now:
# 1. Validate dataset format and completeness
# 2. Convert to training format (LeRobot v3 → GR00T v2.1)
# 3. Launch training job (SageMaker or EC2 spot)
# 4. Monitor progress and report loss
# 5. Deploy checkpoint to inference endpoint
agent("Train a GR00T model on my pick_place_cube dataset")
```

## Implementation sketch

| Component | Responsibility |
|-----------|---------------|
| `tools/training.py` | Agent tool with actions: `prepare_dataset`, `launch`, `status`, `deploy` |
| `training/format_converter.py` | LeRobot v3 → GR00T v2.1 (modality.json, annotations, schema) |
| `training/aws_backend.py` | EC2/SageMaker job lifecycle (provision, monitor, teardown) |
| `training/local_backend.py` | Local GPU training (for Jetson/workstation users) |
| `training/checkpoints.py` | S3 upload/download, deployment to inference server |

## AWS services mapping

| Step | Service |
|------|---------|
| Dataset storage | S3 |
| Format conversion | SageMaker Processing or Lambda |
| GPU training | SageMaker Training Job or EC2 Spot (g6e.xlarge) |
| Checkpoint storage | S3 |
| Inference deployment | EC2 + ECR (existing `gr00t_inference` tool) |
| Connectivity | SSM (no ports to open) |

## Acceptance criteria

- [ ] Agent tool can validate a LeRobot dataset is suitable for GR00T fine-tuning
- [ ] Automatic format conversion (LeRobot v3 → GR00T training format) with zero manual steps
- [ ] Launch training on AWS with a single agent command
- [ ] Progress monitoring (loss curve, ETA)
- [ ] Checkpoint deployment to inference server
- [ ] Works end-to-end: record demos → train → run autonomous — no manual intervention
- [ ] Cost estimates surfaced to user before launch ("This will cost ~$5 for 10k steps on L40S")
- [ ] Documentation covering GPU requirements, expected training times, data quantity guidance

## Context

This was built manually for an SO-100 project using GR00T N1.7:
- 146 episodes, 4 tasks, 10k training steps
- Training time: ~2.5 hours on L40S
- Cost: ~$5
- Final loss: 0.022
- The format conversion and infrastructure setup took longer than the actual training

**Comment by cagataycali** [comment]
We can have full coverage for training pipelines (not for only GR00T), imagining that lerobot + gr00t + ... to cover all training tasks 🤔

**Comment by cagataycali** [comment]
Tracked in #202 (Epic: Training & dataset pipeline).
